var/spack/repos/builtin/packages/wannier90/package.py

```diff
@@ -61,10 +88,68 @@ def edit(self, spec, prefix):
         for key, value in substitutions.items():
             filter_file(key, value, self.makefile_name)
 
-    def install(self, spec, prefix):
+        if '@:2 +shared' in self.spec:
+            # this is to build a .shared wannier90 library
+            filter_file('LIBRARY = ../../libwannier.a',
+                        'LIBRARY = ../../libwannier.' + dso_suffix,
+                        join_path(self.stage.source_path, 'src/Makefile.2'))
+            filter_file('parameters.o kmesh.o io.o comms.o '
+                        'utility.o get_oper.o constants.o '
+                        'postw90_common.o wan_ham.o spin.o '
+                        'dos.o berry.o kpath.o kslice.o '
+                        'boltzwann.o geninterp.o',
+                        'comms.o get_oper.o postw90_common.o '
+                        'wan_ham.o spin.o dos.o berry.o '
+                        'kpath.o kslice.o boltzwann.o geninterp.o',
+                        join_path(self.stage.source_path,
+                                  'src/Makefile.2'))
+            filter_file('../../wannier90.x: .*',
```

Comment:
Is this `.*` actually present in the source code, or do you want to expand to match any string? If it's the latter, you'll need to remove the `string=True` below.

Reply:
Thank you!  You are correct, it should be treated as a regexp!

Reply:
Oops!  I introduced a bug with some missing whitespace!  I will fix it soon and push that fix!  (the build step failed and I'm fairly certain this is the culprit)